### DISCUSSION TITLE
`aggregationRaw` `$match` stage doesn't match ObjectIds

### DISCUSSION THREAD

**Discussion by bn-piar** [prisma/prisma#15793] [Q&A] [answered]
Hello there! Consider the following models: 

```javascript 

model Product {
  collection Collection @relation(fields: [collectionId], references: [id])
  collectionId String @db.ObjectId
}

model Collection {
  products Product[]
}
```
Say I want to execute raw aggregation query that would match certain `Collection`
```javascript

const collections = await prisma.collection.aggregateRaw({
  pipeline: [
    {
      $match: {
        id: {
          $in: products.map(({ collectionId }) => collectionId)
        }
      }
    }
  ]
});
```
it wouldn't find anything because, as I understand, the `collectionId` in `products` array is not an `ObjectId` but a `string`. 

It matches perfectly when I use other unique values that are not `ObjectId`. 

I tried converting `collectionId` to `ObjectId` but then it yells that it doesn't want `ObjectId`, it wants `string`.

**Comment by nurul3101** [comment]
Hello @bn-piar 👋 
I was able to find a GitHub Issue for this: [#15013](https://github.com/prisma/prisma/issues/15013). Could you have a look?
This is on the team's radar and we will be working to get this issue fixed.

**Comment by bn-piar** [comment, marked as answer]
Hey @nurul3101 
I think it's fixed now, at least I can find documents by their `_id`

**Reply by nurul3101** [reply]
Hey @bn-piar , thanks for letting me know, the GitHub Issue is still open, so was it fixed for you by upgrading to the latest version of Prisma?

**Reply by bn-piar** [reply]
@nurul3101 after upgradiing to the latest version the error is gone, yes

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
